prov/tcp/src/tcpx_shared_ctx.c

```diff
@@ -310,6 +310,53 @@ tcpx_match_tag_addr(struct tcpx_rx_ctx *srx, struct tcpx_ep *ep, uint64_t tag)
 	return NULL;
 }
 
+static bool
+tcpx_srx_cancel_rx(struct tcpx_rx_ctx *srx, struct slist *queue, void *context)
+{
+	struct slist_entry *cur, *prev;
+	struct tcpx_xfer_entry *xfer_entry;
+
+	assert(fastlock_held(&srx->lock));
+
+	slist_foreach(queue, cur, prev) {
```

Comment:
Is this macro safe when slist_remove() is involved?

Reply:
If we continued with the iteration, no, but we exit after matching.